model/patch_lifecycle.go

```diff
@@ -438,9 +438,61 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
 		return nil, errors.WithStack(err)
 	}
 
+	if p.IsParent() {
+		//finalize child patches that have a parentStatus = ""
+		for _, childPatch := range p.Triggers.ChildPatches {
+			err = finalizeChildPatch(ctx, childPatch, requester, githubOauthToken)
+			grip.Error(message.WrapError(err, message.Fields{
+				"message":       "unable to finalize child patch",
+				"childPatch_id": childPatch,
+				"patch_id":      p.Id,
+				"variants":      p.BuildVariants,
+				"tasks":         p.Tasks,
+				"variant_tasks": p.VariantsTasks,
+				"alias":         p.Alias,
+			}))
+			return nil, err
+		}
+	}
+
 	return patchVersion, nil
 }
 
+func finalizeChildPatch(ctx context.Context, childPatchId, requester, githubOauthToken string) error {
+	// if the parentStatus is "", finalize without waiting for the parent patch to finish running
+	childPatchDoc, err := patch.FindOneId(childPatchId)
+	if err != nil {
+		return errors.Errorf("error fetching child patch: %s", err)
+	}
+	if childPatchDoc == nil {
+		return errors.New("child patch not found")
+	}
+
+	intent, err := patch.FindIntent(childPatchId, patch.TriggerIntentType)
+	if err != nil {
+		return errors.Errorf("error fetching child patch intent: %s", err)
+	}
+	if intent == nil {
+		return errors.New("childpatch intent not found")
+	}
+
+	if intent.GetParentStatus() == "" {
+		if _, err := FinalizePatch(ctx, childPatchDoc, requester, githubOauthToken); err != nil {
+			grip.Error(message.WrapError(err, message.Fields{
```

Comment:
I think this grip.Error is superfluous: if we return an error it will get grip.Error'ed by the caller.

Reply:
Not necessarily, because one provides the child patch info and one provides the parent patch info. In order to consolidate it, I'd need an additional query in order to provide all the information currently provided.

Reply:
(alternatively, I can include just the parent id, and anyone looking at the logs can query for it)

Reply:
That makes sense.
I like the idea to just include the parent/child ids, but it's up to you

Reply:
done